test/csharp/test_PowerShell.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Management.Automation;
+using Xunit;
+
+namespace PSTests.Parallel
+{
+    public class PowerShellTests
+    {
+        [Fact]
+        public async System.Threading.Tasks.Task TestPowerShellInvokeAsync()
+        {
+            using (PowerShell ps = PowerShell.Create())
```

Comment:
The API's look fine as far as they go. It would probably be (marginally) more efficient to implement them directly rather than on top of BeginInvoke/EndInvoke but that doesn't affect the APIs design. Mostly I'd like to see more tests. What happens if the PowerShell object is created with "CurrentRunspace"? Or is assigned a runspace that is busy? What happens if there's an error in the command or script? I've actually experimented a bit and I think that the behaviours are pretty reasonable but I'd like to see tests for these cases. I'd also like to see test cases in PowerShell as well as C# given that these APIs will also be used from script e.g.:
```powershell
$r1 = [powershell]::Create().AddScript('1..10 | % { [console]::WriteLine("#1 running $_"); sleep 1 }').InvokeAsync()
$r2 = [powershell]::Create().AddScript('1..15 | % { [console]::WriteLine("#2 running $_"); sleep 1 }').InvokeAsync()
[system.threading.tasks.task]::waitall($r1, $r2); write-host "All tasks done"
```
Overall I think this is pretty cool given that, because the APIs return tasks, we we'll be able to use the `Task` methods to synchronize concurrent pipelines, etc. I think that adding these APIs is a definite plus for PowerShell over all so thanks to @KirkMunro for doing this work.

Reply:
Thank you @BrucePay for this feedback. Now that I have your approval, I'll add the remaining unit tests to the PR as requested.

Reply:
Just a status update on this: I have flushed out a bunch of PowerShell Pester tests for these changes, and will now put the complimentary C# tests in place.